packages/image_picker/image_picker/ios/Classes/FLTImagePickerPlugin.m

```diff
@@ -167,12 +170,9 @@ - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result
     ];
     _imagePickerController.videoQuality = UIImagePickerControllerQualityTypeHigh;
 
-    self.result = result;
-    _arguments = call.arguments;
-
-    int imageSource = [[_arguments objectForKey:@"source"] intValue];
-    if ([[_arguments objectForKey:@"maxDuration"] isKindOfClass:[NSNumber class]]) {
-      NSTimeInterval max = [[_arguments objectForKey:@"maxDuration"] doubleValue];
+    int imageSource = [[self.arguments objectForKey:@"source"] intValue];
+    if ([[self.arguments objectForKey:@"maxDuration"] isKindOfClass:[NSNumber class]]) {
```

Comment:
Please use `call.arguments`. It's far less fragile because then there's no order dependency with other code.